src/components/commons/Button/Button.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactNode } from "react";
+import * as S from "./Button.styled";
+
+export interface ButtonPropTypes {
+  size: "xlarge" | "large" | "medium" | "small" | "xsmall";
+  isDisabled?: boolean;
+  varient: "primary" | "line" | "gray" | "blue";
+  children: ReactNode; //자식 노드(텍스트)를 받기 위해 필수.
+  onClick: React.MouseEventHandler<HTMLButtonElement> | undefined; //MouseEvent와는 상이
+}
+
+const Button = ({ onClick, size, isDisabled, varient, children }: ButtonPropTypes) => {
+  return (
+    <>
+      {varient === "primary" && (
+        <S.PrimaryButton
+          onClick={isDisabled ? undefined : onClick}
+          size={size}
+          isDisabled={isDisabled}
+        >
```

Comment:
p3)여기도 disabled 속성이 있으면 onClick이 실행되지는 않아서, 아래처럼 작성해도 될 것 같아요!
```suggestion
        <S.PrimaryButton
          onClick={onClick}
          size={size}
        >
```

Reply:
반영했습니다 ! 다만 제가 아직 실력 부족 이슈로 헷갈릴까봐 styled-components의 이름은 $disabled가 아닌 $isDisabled 로 유지할게요,, 감사합니다 !!

Reply:
넵 확인했습니다 ㅎㅎ